src/modm/driver/inertial/ixm42xxx_data.hpp

```diff
@@ -0,0 +1,269 @@
+// coding: utf-8
+// ----------------------------------------------------------------------------
+/*
+ * Copyright (c) 2023, Rasmus Kleist Hørlyck Sørensen
+ *
+ * This file is part of the modm project.
+ *
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+// ----------------------------------------------------------------------------
+
+#ifndef MODM_IXM42XXX_DATA_HPP
+#define MODM_IXM42XXX_DATA_HPP
+
+#include <cstring>
+#include <span>
+
+#include <modm/math.hpp>
+#include <modm/math/utils/endianness.hpp>
+
+namespace modm
+{
+
+template < class Transport >
+class Ixm42xxx;
+
+namespace ixm42xxxdata
+{
+
+/// @ingroup modm_driver_ixm42xxx
+struct
+Data
+{
+    template < class Transport >
+    friend class ::modm::Ixm42xxx;
+
+    Data(std::span<uint8_t> fifoBuffer = std::span<uint8_t>()) :
+        fifoCount(0),
+        fifoBuffer(fifoBuffer),
+        accelScale(16.f),
+        gyroScale(2000.f)
+    {
+    }
+
+    // DATA ACCESS
+    ///@{
+    int16_t inline
+    getTemp()
```

Comment:
Shouldn't all these acessors be `const`?
```suggestion
    getTemp() const
```

Reply:
Yes I agree!